### ISSUE TITLE
Add support for Pydantic V2

### ISSUE THREAD

**Issue by stephen37** [SeldonIO/MLServer#1419]
MLServer only supports Pydantic V1, which is a problem for us as we would like to move to Pydantic V2 for all our services using Pydantic. 

Do you think MLServer could support it?

**Comment by adriangonz** [comment]
Hey @stephen37 ,

That's a great suggestion - thanks for bringing it up.

Good news are that there's already some WIP to update to Pydantic V2. Although it's lower priority than other stuff, so it may take a while to complete.

**Comment by stephen37** [comment]
Nice to hear that, is there a specific branch where the work is happening? I am asking as I could share it with some of our engineers if they want to help as well for that.

**Comment by dolshevsk** [comment]
Any updates on this? Also waiting for this one to be released 🙏

**Comment by moefsadek** [comment]
Hi @adriangonz,
any updates on this issue ? 
Unfortunately as much as we want to use seldon, we are currently blocked by adopting the mlserver due to this issue, would be great if this can be added soon 🙏

**Comment by adriangonz** [comment]
Hey @moefsadek ,

I'm no longer working in Seldon, but @sakoush has taken over the reins of the MLServer project - so it may be worth checking with him on what's the status of this one.

**Comment by ramonpzg** [comment]
Hi @stephen37, @dolshevsk and @moefsadek -- We are currently working on the migration to Pydantic v2, and just need to finalize some dependency conflicts with other tools our runtimes depend on. We know about the speed improvements and the cool new features Pydantic v2 has to offer so we look forward to moving MLServer to it. We should have an update on this soon.

**Comment by mkurovski1** [comment]
Hey @ramonpzg ,
Thanks for the work you're putting into this 🙏 
I was on my way straight to dependency hell today as well and the pydantic-v1-requirement of MLServer was also involved in that endeavor ;)
Sorry that I have to ask again, but can you already share when we might expect this?
Thanks again!

**Comment by privatedumbo** [comment]
Hey @ramonpzg ! Would you consider something like this in the meantime?
https://github.com/SeldonIO/MLServer/pull/1651

This would make `MLServer` compatible with both versions of Pydantic.

**Comment by jesse-c** [comment]
@privatedumbo: Thank you for your suggestion! It's on our (my) radar to have a look at.

**Comment by jesse-c** [comment]
I've merged in support for Pydantic v2. We don't have a timeline for the next release that will have it, but we're actively workin on it.
